app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,103 @@
+class CoursesController < ApplicationController    
+    def index
+        @user = User.find(params[:user_id])
+        proper_user(@user)
```

Comment:
Is it worth changing the name of this function to check_proper_user() or ensure_authorized_user() or something like that? I feel like it might make the code a bit more self-explanatory.

Reply:
Good idea, I will make this change.